### PR TITLE
enable byte-based regular expressions #229

### PR DESCRIPTION
**Pull request by wash2**
This patch aims to provide support for byte-based regular expressions as discussed in #229 

I used the example provided as a test, but I am open to adding more tests if need be. I would also appreciate any constructive criticism.

### CODE REVIEW

**Comment by wash2** [comment]
It looks like CI succeeded for beta and nightly, but failed on stable (1.32.0) because of an error when installing mdbook. It's unclear to me why this would be related to the changes I've made.

**Comment by Marwes** [comment]
Updating to 1.36 as minimum on CI should fix it https://doc.rust-lang.org/std/mem/union.MaybeUninit.html . Looks like `SmallVec` broke its minimum supported rust version.

File: lalrpop-test/src/match_section_byte.lalrpop [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    UPDATE => String::from("UPDATE")
+};
+
+Table: String = {
+    <r"(?i-u)[a-z]+"> => String::from(<>)
+};
+
+pub Query: String = {
+    <Keyword> <Table> => format!("{} {}", <>)
+};
```
Comment (Marwes) [inline lalrpop-test/src/match_section_byte.lalrpop]: Nit: Missing newline

**Comment by wash2** [comment]
> Updating to 1.36 as minimum on CI should fix it https://doc.rust-lang.org/std/mem/union.MaybeUninit.html . Looks like `SmallVec` broke its minimum supported rust version.

I'm kinda new to using CI. How can I set 1.36 as minimum?

File: lalrpop-test/src/match_section_byte.lalrpop (hunk 1 above)
Comment (wash2) [reply lalrpop-test/src/match_section_byte.lalrpop]: just added the newline.

**Comment by Marwes** [comment]
Nice!

**Pull request merged**